docs/upgrade.html

```diff
@@ -227,6 +227,11 @@ <h5><a id="upgrade_400_notable" href="#upgrade_400_notable">Notable changes in 4
                     Java 8 support has been removed in Apache Kafka 4.0
                     See <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=181308223">KIP-750</a> for more details
                 </li>
+                <li>
+                    Logging framework has been migrated from Log4j to Log4j2.
```

Comment:
Shall we mention the behavior if you do _not_ convert the configuration files? I believe there is a compatibility mode with some limitations?

Reply:
Log4j provides a way to convert Log4j 1.x properties at runtime ([Log4j Migration Guide](https://logging.apache.org/log4j/2.x/migrate-from-log4j1.html#ConfigurationCompatibility)).

To enable this feature, we can set both `LOG4J_COMPATIBILITY=true` and specify the configuration file using `LOG4J_CONFIGURATION_FILE=file`.

Reply:
Do we want to automatically detect if log4j 1.x properties are used and, if so, automatically enable this mode? It seems like a huge problem if customers lose their logging while upgrading to 4.0+.

Reply:
We can always enable that compatibility feature as the bridge jar is already included now.

Reply:
Even better if it can be always enabled with no significant downside.